app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterViewModel.kt

```diff
@@ -20,24 +22,52 @@ class CharacterViewModel(private val characterUseCase: CharacterUseCase) : ViewM
 
     fun fetchCharacters() {
         viewModelScope.launch {
-            mutableCharactersState.value = Event(Data(responseType = Status.LOADING))
-            when (val result = withContext(Dispatchers.IO) { characterUseCase() }) {
+            mutableCharactersState.postValue(Event(Data(responseType = Status.LOADING)))
+            when (val result: Result<CharacterDataWrapperEntity> = withContext(Dispatchers.IO) { characterUseCase() }) {
                 is Result.Success -> {
                     if (result.data.data.results.isNotEmpty()) {
-                        mutableCharactersState.value = Event(Data(responseType = Status.SUCCESSFUL, data = result.data))
+                        mutableCharactersState.postValue(Event(Data(responseType = Status.SUCCESSFUL, data = result.data)))
                     } else {
-                        mutableCharactersState.value = Event(Data(responseType = Status.EMPTY_LIST, data = result.data))
+                        mutableCharactersState.postValue(Event(Data(responseType = Status.EMPTY_LIST, error = NO_CHARACTERS_MESSAGE)))
                     }
                 }
                 is Result.Failure -> {
-                    mutableCharactersState.value = Event(Data(responseType = Status.ERROR, error = result.exception.message))
+                    result.exception.message?.let {
+                        mutableCharactersState.value = Event(Data(responseType = Status.ERROR, error = it))
+                    }
                 }
             }
         }
 
     }
+
+    companion object {
+        private const val NO_CHARACTERS_MESSAGE: String = "No characters found"
+    }
 }
 
-data class Data<RequestData>(var responseType: Status, var data: RequestData? = null, var error: String? = null)
+data class Data<CharacterDataWrapperEntity>(
+    var responseType: Status,
+    var data: com.example.domain.entity.CharacterDataWrapperEntity = CharacterDataWrapperEntity(
+        code = ConstantUtils.INT_DEFAULT,
+        status = ConstantUtils.STRING_DEFAULT,
+        copyright = ConstantUtils.STRING_DEFAULT,
+        attributionText = ConstantUtils.STRING_DEFAULT,
+        attributionHTML = ConstantUtils.STRING_DEFAULT,
+        data = CharacterDataContainerEntity(
+            offset = ConstantUtils.INT_DEFAULT,
+            limit = ConstantUtils.INT_DEFAULT,
+            total = ConstantUtils.INT_DEFAULT,
+            count = ConstantUtils.INT_DEFAULT,
+            results = emptyList()
+        ),
```

Comment:
these default values should be in CharacterDataContainerEntity

Reply:
done